### PR TITLE
CASMTRIAGE-7380 -- HSN interfaces names have changed and are now incorrect

### PR DESCRIPTION
**Pull request by jpdavis-prof**
# Description

CASMTRIAGE-7380
-  https://jira-pro.it.hpe.com:8443/browse/CASMTRIAGE-7380

# Checklist
- [x] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [x] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

### CODE REVIEW

File: upgrade/scripts/upgrade/prerequisites.sh [hunk 1]
```diff
@@ -1160,6 +1160,46 @@ if [[ ${state_recorded} == "0" && $(hostname) == "${PRIMARY_NODE}" ]]; then
...
+
+    # Loop through one at a time. If `--hosts` isn't provided, we will error out on the 'Global' key.
+    for ncn_xname in "${NCN_XNAMES[@]}"; do
+      printf "% -15s: " "${ncn_xname}"
+
+      params=$(cray bss bootparameters list --hosts "${ncn_xname}" --format json | jq '.[] |."params"' \
+        | sed -E \
+          -e 's/ip=hsn[0-9]+:auto6//g' \
+          -e 's/ifname=hsn[0-9]+:[0-9a-fA-F:]{17}\s+//' \
+        | tr -d \")
```
Comment (don-bahls-hpe) [inline upgrade/scripts/upgrade/prerequisites.sh]: I'm curious what the "tr -d \" is doing.

File: upgrade/scripts/upgrade/prerequisites.sh [hunk 2]
```diff
@@ -1160,6 +1160,46 @@ if [[ ${state_recorded} == "0" && $(hostname) == "${PRIMARY_NODE}" ]]; then
       csi handoff bss-update-param --set "${bootparameter}"
     done
 
+    # Get a list of NCNs.
+    if IFS=$'\n' read -rd '' -a NCN_XNAMES; then
+      :
+    fi <<< "$(cray hsm state components list --role Management --type Node --format json | jq -r '.Components | map(.ID) | join("\n")')"
```
Comment (don-bahls-hpe) [inline upgrade/scripts/upgrade/prerequisites.sh]: We could theoretically filter on `--subrole Worker` here, but there's no harm in changing everything I guess.

File: upgrade/scripts/upgrade/prerequisites.sh (hunk 1 above)
Comment (jpdavis-prof) [reply upgrade/scripts/upgrade/prerequisites.sh]: It's removing double quotes. We don't want double quotes at the beginning or end of our params. For the sake of quickness I copied what was in 1.5, but I just updated tr to use sed instead.

File: upgrade/scripts/upgrade/prerequisites.sh [hunk 3]
```diff
@@ -1160,6 +1160,46 @@ if [[ ${state_recorded} == "0" && $(hostname) == "${PRIMARY_NODE}" ]]; then
...
+    error=0
+
+    # Loop through one at a time. If `--hosts` isn't provided, we will error out on the 'Global' key.
+    for ncn_xname in "${NCN_XNAMES[@]}"; do
+      printf "% -15s: " "${ncn_xname}"
+
+      params=$(cray bss bootparameters list --hosts "${ncn_xname}" --format json | jq '.[] |."params"' \
+        | sed -E \
+          -e 's/ip=hsn[0-9]+:auto6//g' \
+          -e 's/ifname=hsn[0-9]+:[0-9a-fA-F:]{17}\s+//' \
```
Comment (rustydb) [inline upgrade/scripts/upgrade/prerequisites.sh]: Two issues.

1. Remove the trailing white space when it exists (when this isn't the final parameter in the list). 1183 was not trimming whitespace, and 1184 was greedily trimming white space. Use `\s?`.
2. The new regex only replaces the first `ifname=hsn`, it'll miss one if two are present (hsn0 and hsn1). Add the `/g` (global) flag in the new regex
```suggestion
          -e 's/ip=hsn[0-9]+:auto6\s?//g' \
          -e 's/ifname=hsn[0-9]+:[0-9a-fA-F:]{17}\s?//g' \
```

**Review by rustydb — CHANGES_REQUESTED** [review]

File: upgrade/scripts/upgrade/prerequisites.sh (hunk 2 above)
Comment (jpdavis-prof) [reply upgrade/scripts/upgrade/prerequisites.sh]: I assumed that param should not exist on all NCN's. I can switch it if needed / it may break something.

Comment (rustydb) [reply upgrade/scripts/upgrade/prerequisites.sh]: To Don's point, `hsn` NICs only exist on workers so it's just less to process looking at. However, it doesn't hurt to scrub every NCN.

**Review by rustydb — APPROVED** [review]

File: upgrade/scripts/upgrade/prerequisites.sh (hunk 1 above)
Comment (don-bahls-hpe) [reply upgrade/scripts/upgrade/prerequisites.sh]: OK that makes sense.   I was overthinking what that was doing.

**Review by don-bahls-hpe — APPROVED** [review]

**Review by nrockershousen — APPROVED** [review]

**Pull request merged**